src/lib/RepositoryForms/Form/Processor/Content/UrlRedirectProcessor.php

```diff
@@ -0,0 +1,141 @@
+    private $requestStack;
+
+    /** @var \eZ\Publish\API\Repository\URLAliasService */
+    private $URLAliasService;
+
+    /** @var array */
+    private $siteaccessGroups;
+
+    /**
+     * @param \Symfony\Component\Routing\RouterInterface $router
+     * @param \Symfony\Component\HttpFoundation\RequestStack $requestStack
+     * @param \eZ\Publish\API\Repository\URLAliasService $URLAliasService
+     * @param array $siteaccessGroups
+     */
+    public function __construct(
+        RouterInterface $router,
+        RequestStack $requestStack,
+        URLAliasService $URLAliasService,
+        array $siteaccessGroups
+    ) {
+        $this->router = $router;
+        $this->requestStack = $requestStack;
+        $this->URLAliasService = $URLAliasService;
+        $this->siteaccessGroups = $siteaccessGroups;
+    }
+
+    /**
+     * @return array
+     */
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            RepositoryFormEvents::CONTENT_PUBLISH => ['processRedirectAfterPublish', 0],
+            RepositoryFormEvents::CONTENT_CANCEL => ['processRedirectAfterCancel', 0],
+        ];
+    }
+
+    /**
+     * @param \EzSystems\RepositoryForms\Event\FormActionEvent $event
+     *
+     * @throws \EzSystems\EzPlatformAdminUi\Exception\InvalidArgumentException
+     */
+    public function processRedirectAfterPublish(FormActionEvent $event): void
+    {
+        if (!$this->isAdminSiteAccess()) {
+            return;
+        }
+
+        if ($event->getForm()['redirectUrlAfterPublish']->getData()) {
+            return;
+        }
+
+        /** @var \Symfony\Component\HttpFoundation\RedirectResponse $response */
+        $response = $event->getResponse();
+
+        if (!$response instanceof RedirectResponse) {
+            return;
+        }
+
+        $targetUrlAlias = $this->URLAliasService->lookup(
+            $response->getTargetUrl()
+        );
+
+        $response = new RedirectResponse($this->router->generate(
+            '_ezpublishLocation',
+            ['locationId' => $targetUrlAlias->destination],
+            UrlGeneratorInterface::ABSOLUTE_URL
+        ));
+        $event->setResponse($response);
+    }
+
+    /**
+     * @param \EzSystems\RepositoryForms\Event\FormActionEvent $event
+     *
+     * @throws \EzSystems\EzPlatformAdminUi\Exception\InvalidArgumentException
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     */
+    public function processRedirectAfterCancel(FormActionEvent $event): void
+    {
+        if (!$this->isAdminSiteAccess()) {
+            return;
+        }
+
+        /** @var \Symfony\Component\HttpFoundation\RedirectResponse $response */
+        $response = $event->getResponse();
+
+        if (!$response instanceof RedirectResponse) {
+            return;
+        }
+
+        $targetUrlAlias = $this->URLAliasService->lookup(
```

Comment:
Can you make it fail safe? Check if the alias can be loaded first as someone could change the redirection in his processor to something different than Url Alias.